pkg/rbac/sharedData.go

```diff
@@ -210,3 +221,24 @@ func (shared *SharedData) GetManagedClusters(cache *Cache, ctx context.Context)
 	return shared.mcErr
 
 }
+
+func (cache *Cache) GetDisabledClusters() (*map[string]struct{}, error) {
+
+	if cache.SharedCacheDisabledClustersValid() {
+		klog.V(5).Info("Search Addon DisabledClusters Cache valid")
+	} else {
+		klog.V(5).Info("DisabledClusters Cache not valid") // - running query to get search addon disabled clusters")
+		cache.SetDisabledClusters(map[string]struct{}{}, fmt.Errorf("addon disabled clusters cache is invalid"))
+	}
+	cache.shared.dcLock.Lock()
+	defer cache.shared.dcLock.Unlock()
+	return &cache.shared.disabledClusters, cache.shared.dcErr
+}
+
+func (cache *Cache) SetDisabledClusters(disabledClusters map[string]struct{}, err error) {
```

Comment:
Keep this function private (lowercase).  Data should only get set within this package.
```suggestion
func (cache *Cache) setDisabledClusters(disabledClusters map[string]struct{}, err error) {
```

Reply:
This change might trigger some refactoring because I see `SetDisabledClusters()` is getting called from messages.go

Ideally `messages.go` would just request the disabled clusters and it should be up to the cache logic to return existing cached data or trigger a request to refresh the data.